### DISCUSSION TITLE
How to cancel/terminate a RpcOutputStream from the client side?

### DISCUSSION THREAD

**Discussion by swwilson-bsi** [timostamm/protobuf-ts#179] [Q&A] [answered]
When a client calls a server method that returns a stream, how can the client tell the server to stop writing data to the stream?

**Comment by caesuric** [comment, marked as answer]
With the caveat that I haven't tested it yet since it's a work in progress, according to the manual, it appears that this is the correct way:
```
const aborter = new AbortController();
const stream = this.client.streamingCall(message, {
    abort: aborter.signal
});

// cancel later
aborter.abort();
```

**Reply by swwilson-bsi** [reply]
Thanks. That worked for me.

**Reply by timostamm** [reply]
> With the caveat that I haven't tested it yet since it's a work in progress, according to the manual

This feature should not be WIP. Can you point me to the location in the manual?

**Reply by caesuric** [reply]
> This feature should not be WIP. Can you point me to the location in the manual?

Sorry, I didn't phrase that well. The FEATURE is not a work in progress, my CODE at that time was a work in progress and not ready to test as I was converting a large codebase over to use this library and needed to finish to really test the system well.
